code/src/reducers/gamestate.js

```diff
@@ -0,0 +1,77 @@
+import { createSlice } from '@reduxjs/toolkit';
+// import { ui } from './ui'
+
+export const gameState = createSlice({
+  name: 'gameState',
+  initialState: {
+    username: '',
+    stage: []
+  },
+  reducers: {
+    setUser: (store, action) => {
+      store.username = `${new Date().getTime()}+${action.payload}`
+      store.showedUsername = action.payload
+    },
+    setStage: (store, action) => {
+      store.stage = action.payload
+    },
+    setRestartGame: (state) => {
+      state.username = '';
+      state.stage = [];
+    },
+    setLoading: (store, action) => {
+      store.isLoading = action.payload;
+    }
+  }
+});
+
+export const fetchGame = () => {
+  return (dispatch, getState) => {
+    dispatch(gameState.actions.setLoading(true))
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ username: getState().gameState.username })
+    };
+
+    fetch('https://labyrinth.technigo.io/start', options)
```

Comment:
Another approach is to create a variable with the https and use it inside the fetch. Same as u did with `const options` ex:
`const API_URL = 'https://labyrinth.technigo.io/start' `
`fetch(API_URL, options)`
You could even create a new file named utils and declare all those variables there and then import them